src/nvim/eval.c

```diff
@@ -9921,9 +9906,6 @@ static void f_getcwd(typval_T *argvars, typval_T *rettv)
 #ifdef BACKSLASH_IN_FILENAME
   slash_adjust(rettv->vval.v_string);
 #endif
-
-end:
-  xfree(cwd);
```

Comment:
Not sure it's a good idea to remove this--if more cleanup is needed for future changes, someone needs to re-discover the lines that were changed to `return` in this diff.